lib/views/item-scene.js

```diff
@@ -22,6 +22,49 @@ class ItemScene extends React.Component {
   constructor(props) {
     super(props);
     debug('created');
+
+    let distance = this.props.item.distance;
+    if (typeof distance !== 'number') {
+      distance = null;
+    }
+
+    this.state = {
+      distance,
+    };
+  }
+
+  componentDidMount() {
+    if (this.state.distance !== null ||
+      typeof this.props.item.latitude !== 'number' ||
+      typeof this.props.item.longitude !== 'number') {
+      return;
+    }
+
+    navigator.geolocation.getCurrentPosition((pos) => {
```

Comment:
I think it makes sense that the components that depend on location call `startLocationUpdates(actions.locationChange)` when they are created and `stopLocationUpdates(action.locationChange)` when they are destroyed.

If more than one component needs location updates at the same time then we'll need to keep track of number of 'clients' bound so we know whether we can stop.

Happy to discuss this more is need be :)